src/openrct2/entity/Peep.cpp

```diff
@@ -1611,7 +1612,7 @@ void Peep::FormatNameTo(Formatter& ft) const
             ft.Add<StringId>(_staffNames[staffNameIndex]);
             ft.Add<uint32_t>(PeepId);
         }
-        else if (GetGameState().Park.Flags & PARK_FLAGS_SHOW_REAL_GUEST_NAMES)
+        if (gameState.Park.Flags & PARK_FLAGS_SHOW_REAL_GUEST_NAMES)
```

Comment:
Probably should keep this as an `else if`.

Reply:
Yeah, I think you should start with the real guest name here, maybe:
```c++
if (gameState.Park.Flags & (PARK_FLAGS_SHOW_REAL_GUEST_NAMES || PARK_FLAGS_SHOW_REAL_STAFF_NAMES))
{
    auto realNameStringId = GetRealNameStringIDFromPeepID(PeepId);
    ft.Add<StringId>(realNameStringId);
}
else if (auto* staff = As<Staff>(); staff != nullptr)
{
    // ...
}
else
{
    ft.Add<StringId>(STR_GUEST_X).Add<uint32_t>(PeepId);
}
```

Reply:
While I like the looks of that, it doesn't quite work, unfortunately. You need an explicit check for staff type, otherwise you'd get a real name for staff with just `PARK_FLAGS_SHOW_REAL_GUEST_NAMES` set.